src/dsp_tools/analyse_xml_data/extract_links_from_XML.py

```diff
@@ -93,136 +103,239 @@ def _make_graph(
     """
```

Comment:
The docstring is a bit misleading:

```
xml_instances: Instances of links to texts containing links to other resources
```

The links are not TO texts, but IN texts

Reply:
Well if we look at it how it will be in the triplestore, I disagree. I found it very awkward to describe these things as the representation in the graph is not the same as in the triplestore. But still we have to understand how the XMLLinks differ from ResptrLinks when we work with the rx graph. At this point would you focus on how it is in the rx represenation or what it represents in the triplestore?

Reply:
Technically there is no link relationship between the starting resource and its text values; the text values are related to their containing resource by a property, but for text values this is not a link property but a regular value property pointing to a TextValue.  
And although this may seem like hair-splitting: DSP-TOOLS does not know anything about the existence of a triplestore, it only knows of an arbitrary API which we expose, and which may or may not directly correspond to the model and technology we use to persist the data behind this API. And, in fact, this is a striking place of where this is not the case: The API exposes XML, but in the triplestore we persist the markup as standoff.  
And on top of that, even "resptr" is a concept that is exclusive to the DSP-TOOLS upload XML and is used nowhere else in our system.  
I think it's important to keep these distinct layers separated in our minds.

As for this docstring: This function is where the XML model is turned in the Graph model, so the goal should probably be to make clear how the XML maps to the graph model.

Here is a suggestion:

```
    Args:
        resptr_links: A list of objects representing a direct link (resptr) between a starting resource and a target resource
        xml_links: A list of objects representing one or more links from a single starting resource to a set of target resources, where all targed resources are linked to from a single text value on the starting resource.
        all_resource_ids: IDs of all the resources in the graph
```

I'm sure this has room for improvement (and, admittedly, it's a mouth full!), but I think it captures the inherent logic and stresses the differences between the two kinds of link-objects at hand.

Reply:
Thanks, I am happy for your input, as I said I was unsure myself how describe it.